src/utils/hooks/useTriggerProcessing.tsx

```diff
@@ -0,0 +1,28 @@
+import { useState, useEffect } from 'react'
+import { useGetTriggersQuery } from '../../utils/generated/graphql'
+
+type triggerTypes = 'applicationTrigger' | 'reviewTrigger'
+
+const useTriggerProcessing = (props: { serialNumber: string; trigger: triggerTypes }) => {
+  const { serialNumber, trigger } = props
+  const [isProcessing, setIsProcessing] = useState(true)
+
+  const { data, loading, error } = useGetTriggersQuery({
+    variables: {
+      serial: serialNumber,
+    },
+    pollInterval: 500,
+    skip: !isProcessing,
+    fetchPolicy: 'no-cache',
+  })
+
+  useEffect(() => {
+    if (data?.applicationTriggerStates?.nodes[0]) {
```

Comment:
Probably worth checking for unexpected data before doing this!

Reply:
Okay, have added an additional `triggerError` field, which is combined into the apollo error field on return. 

You can test it in action by temporarily removing the `serial` field from the `getTriggers.query.ts`. :)